text/0000-std-rpc-api.md

```diff
@@ -0,0 +1,503 @@
+
+```cs
+void Update()
+{
+    if (Input.GetKeyDown(KeyCode.P))
+    {
+        PongClientRpc(Time.frameCount, "hello, world"); // Server -> Client
+    }
+}
+```
+
+Marking method with `[ClientRpc]` attribute and putting `ClientRpc` suffix to the method name are required, otherwise it will prompt error messages:
+
+```cs
+// Error: Invalid, missing 'ClientRpc' suffix in the method name
+[ClientRpc]
+void Pong(int somenumber, string sometext) { /* ... */ }
+
+// Error: Invalid, missing [ClientRpc] attribute on the method
+void PongClientRpc(int somenumber, string sometext) { /* ... */ }
+```
+
+`[ClientRpc]` attribute and matching `...ClientRpc` suffix in the method name are there to make it crystal clear for RPC call sites to know when they are executing an RPC, it will be replicated and executed on the client-side, without necessarily jumping into original RPC method declaration to find out if it was an RPC, if so whether it is a ServerRpc or ClientRpc:
+
+```cs
+Pong(somenumber, sometext); // Is this an RPC call?
+
+PongRpc(somenumber, sometext); // Is this a ServerRpc call or ClientRpc call?
+
+PongClientRpc(somenumber, sometext); // This is clearly a ClientRpc call
+```
+
+### Reliability
+
+RPCs are reliable by default which means they are guaranteed to be executed on the remote side. However, sometimes developers might want to opt-out reliability, which is often the case for non-critical events such as particle effects, sounds effects etc.
+
+Reliability configuration can be specified for both `ServerRpc` and `ClientRpc` methods at compile-time:
+
+```cs
+[ServerRpc]
+void MyReliableServerRpc() { /* ... */ }
+
+[ServerRpc(IsReliable = false)]
+void MyUnreliableServerRpc() { /* ... */ }
+
+[ClientRpc]
+void MyReliableClientRpc() { /* ... */ }
+
+[ClientRpc(IsReliable = false)]
+void MyUnreliableClientRpc() { /* ... */ }
+```
+
+Reliable RPCs will be received on the remote end in the same order as they are fired but this in-order guarantee only applies to RPCs on the same `NetworkObject`. Different `NetworkObject`s might have reliable RPCs called but executed in different order compared to each other. To put more simply, in-order reliable RPC execution is guaranteed per `NetworkObject` basis only.
+
+An RPC call made without an active connection will be dropped and will not be queued for send automatically. Both reliable and unreliable RPC calls have to be made when there is an active network connection established between a client and the server. Also reliable RPC calls made during connection will be dropped on disconnect as well.
+
+### Execution Table
+
+An RPC function **never** executes its body immediately since it's being a network construct. Even a `ServerRpc` called by a host (an instance that is a client and the server at the same time, aka listen server) will not be executed immediately but follow the regular network frame staging first.
+
+||Server|Client|Host (Server+Client)|
+|-:|:-:|:-:|:-:|
+|ServerRpc Network Send|❌|✅|✅|
```

Comment:
I had to really think about this table.  I think this could be illustrated more clearly maybe more of a ladder / flow diagram where you have the server / client as the 'columns' and have time going vertically